File: java/src/WikiApp.java
class WikiApp {
	WikiPage root_page;

	public WikiApp(WikiPage root_page) {
		this.root_page = root_page;
	}

	public Response handle_request(Request request) {
		ResultResponder responder = ResponderBuilder.responder_for(request);
		return responder.make_response(request, new RequestContext(root_page));
	}
}

File: csharp/WikiApp.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;

namespace WikiSearch
{
	public class WikiApp
	{
		WikiPage rootPage;

		public WikiApp(WikiPage root_page)
		{
			this.rootPage = root_page;
		}

		public Response HandleRequest(Request request) {
			ResultResponder responder = ResponderBuilder.ResponderFor(request);
			return responder.MakeResponse(request, new RequestContext(rootPage));
		}
	}
}

